lib/challenge.js

```diff
@@ -61,6 +91,24 @@ var some_pet;
 var Pet;
 // Answer Starts Here
 
+var Pet = function Pet(name, species, breed, noise) {
+  this.name = name;
+  this.species = species;
+  this.breed = breed;
+  this.noise = []];
+
+};
+
+Pet.prototype.makeNoise = function makeNoise () {
+  var makeNoises = 0;
+  for (var i = 0; i < this.noise.length; i++) {
+    makeNoises += this.noise[i].noise;
+  }
+  return makeNoise;
```

Comment:
Here, that variable name I pointed out earlier bites you. This function doesn't return the variable `makeNoises`, but the function itself, which internally calls itself `makeNoise`!